commit e45498c138c1406122179219f1b54dc88ffbcc62 (refs/changes/31/531/2)
Author: Ryan Zezeski <rpz@joyent.com>
Date:   2016-09-26T15:34:58-04:00 (3 years ago)
    
    OS-5666 statechange lock_file() use of trap could be less confusing

diff --git a/overlay/generic/usr/lib/brand/jcommon/poststate b/overlay/generic/usr/lib/brand/jcommon/poststate
index cde9db9b..4e4b1207 100644
--- a/overlay/generic/usr/lib/brand/jcommon/poststate
+++ b/overlay/generic/usr/lib/brand/jcommon/poststate
@@ -25,7 +25,7 @@ if [[ -n $_ZONEADMD_brand_debug ]]; then
 	echo "zone $1 post-state-change $3 $4" >>$logfile
 	ksh -x /usr/lib/brand/$ps_brand/statechange "post" $@ 2>>$logfile
 	res=$?
-	echo "zone $1 post-state-change result $?" >>$logfile
+	echo "zone $1 post-state-change result $res" >>$logfile
 else
 	/usr/lib/brand/$ps_brand/statechange "post" $@
 	res=$?
diff --git a/overlay/generic/usr/lib/brand/jcommon/prestate b/overlay/generic/usr/lib/brand/jcommon/prestate
index 5fa95496..a61af897 100644
--- a/overlay/generic/usr/lib/brand/jcommon/prestate
+++ b/overlay/generic/usr/lib/brand/jcommon/prestate
@@ -25,7 +25,7 @@ if [[ -n $_ZONEADMD_brand_debug ]]; then
 	echo "zone $1 pre-state-change $3 $4" >>$logfile
 	ksh -x /usr/lib/brand/$ps_brand/statechange "pre" $@ 2>>$logfile
 	res=$?
-	echo "zone $1 pre-state-change result $?" >>$logfile
+	echo "zone $1 pre-state-change result $res" >>$logfile
 else
 	/usr/lib/brand/$ps_brand/statechange "pre" $@
 	res=$?
diff --git a/overlay/generic/usr/lib/brand/jcommon/statechange b/overlay/generic/usr/lib/brand/jcommon/statechange
index 0ad01040..339510fc 100644
--- a/overlay/generic/usr/lib/brand/jcommon/statechange
+++ b/overlay/generic/usr/lib/brand/jcommon/statechange
@@ -85,7 +85,7 @@ lock_file()
 	typeset prev_pid=0
 	while true; do
 		if (set -o noclobber; echo "$$" >$LOCKFILE) 2>/dev/null; then
-			trap 'ecode=$? rm -f $LOCKFILE; exit $ecode' INT TERM EXIT
+			trap 'rm -f $LOCKFILE' EXIT
 			break;
 		fi
 
